cdap-kafka/src/main/java/co/cask/cdap/kafka/run/KafkaServerMain.java

```diff
@@ -136,29 +142,24 @@ public void destroy() {
     // Nothing to do
   }
 
-  private Properties generateKafkaConfig(int brokerId, String zkConnectStr, String hostname, int port,
-                                         int numPartitions, int replicationFactor, String logDir) {
-    Preconditions.checkState(port > 0, "Port number is invalid.");
-    Preconditions.checkState(numPartitions > 0, "Num partitions should be greater than zero.");
+  private int tryParse(String val) {
```

Comment:
Well, parseInt will throw the NumberFormatException, if its not a valid number (so we can remove the catch and this tryParse method).
We'll still need the Preconditions check to ensure its not a negative number.